rfcs/converged-implementation-patterns.md

```diff
@@ -0,0 +1,171 @@
+# RFC: Converged component implementation pattern
+
+<!--
+An RFC can be anything. A question, a suggestion, a plan. The purpose of this template is to give some structure to help folks write successful RFCs. However, don't feel constrained by this template; use your best judgement.
+
+Tips for writing a successful RFC:
+
+- Simple plain words that make your point, fancy words obfuscate
+- Try to stay concise, but don't gloss over important details
+- Try to write a neutral problem statement, not one that motivates your desired solution
+- Remember, "Writing is thinking". It's natural to realize new ideas while writing your proposal
+-->
+
+---
+
+_List contributors to the proposal here_
+
+## Summary
+
+<!-- Explain the proposed change -->
+
+Tries to commit the common component pattern to writing in the repo for future reference and track improvements
+
+Code samples included could possibly be applied to utility scripts to generate packages
+
+## Background
+
+<!-- If there is relevant background include it here -->
+
+Many new engineers can find it hard to adapt to a new codebase, especially when the coding patterns and practices are fairly new and in a constant state of flux
+
+## Problem statement
+
+<!--
+Why are we making this change? What problem are we solving? What do we expect to gain from this?
+
+This section is important as the motivation or problem statement is indepenent from the proposed change. Even if this RFC is not accepted this Motivation can be used for alternative solutions.
+
+In the end, please make sure to present a neutral Problem statement, rather than one that motivates a particular solution
+-->
+
+Many new developers might have questions on the patterns that we follow to write a React component. This RFC attempts to commit and possibbly improve our code patterns in writing so that there might be a reference for new (and old) members of the team.
+
+## Detailed Design or Proposal
+
+<!-- This is the bulk of the RFC. Explain the proposal or design in enough detail for the inteded audience to understand. -->
+
+### Hook based architecture.
+
+Generally a component will have the following different files. Let us consider a `Sample` component
+
+- Sample.tsx
+- useSample.ts
+- useSampleClasses/useSampleStyles.ts
+- renderSample.tsx
+- Sample.types.ts
+
+#### renderSample.tsx
+
+renders the correct JSX output of the component and its slots given the correct state.
```

Comment:
So far, this is going absolutely great! What additionally I'd love to see here is an explanation of what a slot is. Do you think this would make sense to be explained here?

Reply:
```suggestion
Renders the correct JSX output of the component and its slots given the correct state.
```

Just so that we're consistent throughout ;-)

Reply:
This is an interesting one, there's another RFC @layershifter  has made for slots.... I'm not sure if it's worth converging the two or just referencing

It's also in PR state tho :/ 

https://github.com/microsoft/fluentui/pull/16529/files#